add blog about reactive renaming (#1913)
quarkusio/quarkusio.github.io: _posts/2024-03-14-the-big-rename.adoc

---
layout: post
title: 'Evolving Quarkus Extension Naming for Clarity'
date: 2024-04-14
tags: announcement
synopsis: Recently, we've made a decision to rename several Quarkus extensions that previously included the word "reactive" in their names. This blog post aims to shed light on the reasoning behind this change and what it means for you as a user and/or extension writer.
author: maxandersen
---

Quarkus is about providing a modern, efficient, and productive development experience for Java developers. We're committed to making it easier for you to build cloud-native applications, whether you're using traditional blocking, reactive, or virtual thread-based programming models.

> Historically Quarkus has used the term "reactive" in the names of several extensions to indicate their *additional* support for non-blocking, event-driven programming. 

Reviewer: ```suggestion
Historically, Quarkus has used the term "reactive" in the names of several extensions to indicate their *additional* support for non-blocking, event-driven programming. 
```

> However, we've found that this terminology has led to confusion and misunderstanding among our users. Many have mistakenly believed that using these "reactive" extensions forces them into reactive programming, even though Quarkus always by default recommend traditional blocking programming model for your application and only optionally use reactive. 

Reviewer: ```suggestion
However, we've found that this terminology has led to confusion and misunderstanding among our users. Many have mistakenly believed that using these "reactive" extensions forces them into reactive programming, even though Quarkus, by default, recommends the traditional blocking programming model for your application and only use reactive programming if you have needs for it (like high concurrency requirement, or the need to orchestrate many asynchronous tasks). 
```

With the recent production release of Virtual Threads in Java 21, the distinction between traditional, reactive, and now, virtual thread-based programming has become even more critical. 

== Our Solution: A New Naming Strategy

To address this and guide our users more effectively, we've decided to revise the naming convention for certain extensions. This change aims to clarify the capabilities and flexibility of Quarkus extensions, making it easier for you to choose the right extension for your project.

Here's a summary of the changes:

- *Quarkus RESTEasy Reactive* becomes *Quarkus REST*, emphasizing its general-purpose applicability beyond just reactive programming.
- *Quarkus SmallRye Reactive Messaging* is now *Quarkus Messaging*, indicating its broader use cases.
- Extensions that inherently push for a reactive programming model, like *Quarkus Hibernate Reactive*, will retain the "reactive" terminology.
- The *Quarkus REST Client Reactive* is simplified to *Quarkus REST Client*, unifying the naming under a more general "REST" umbrella.

> 

Reviewer: Quarkus REST, Quarkus Messaging and Quarkus REST Client are still leveraging the reactive engine underneath for efficiency and performances. At the user level, this is an implementation details.

== Impact on users

> These changes are designed to make it clearer that you're not constrained to reactive programming when using these extensions. Whether you prefer a traditional blocking approach, reactive programming and/or virtual threads, Quarkus supports you.

Reviewer: ```suggestion
These changes are designed to clarify that you are not constrained to reactive programming when using these extensions. Whether you prefer a traditional blocking approach, reactive programming, or virtual threads, Quarkus supports you.
```

> These changes are designed to make it clearer that you're not constrained to reactive programming when using these extensions. Whether you prefer a traditional blocking approach, reactive programming and/or virtual threads, Quarkus supports you.

Reviewer 2: ```suggestion
These changes are designed to make it clearer that you're not constrained to reactive programming when using these extensions. Whether you prefer a traditional blocking approach, reactive programming and/or virtual threads, Quarkus has you covered.
```

For most projects, this renaming should not require changes, except for those having used `quarkus-rest-client` it may require some adjustments in your dependencies. We're committed to making this transition as smooth as possible. Here are a few key points to help you adapt:

> - **Maven Relocations:** We'll introduce Maven relocations for the renamed extensions, guiding you towards the new names when you update your dependencies.

Reviewer: ```suggestion
- **Maven Relocations:** We'll introduce Maven relocations for the renamed extensions, guiding you toward the new names when you update your dependencies.
```

- **Tooling Updates:** The Quarkus CLI will default to use the new names for new projects but continue to work with old names. `quarkus update` will get rules to help perform the update/renames if/when relevant.

> - **Documentation:** We'll update our documentation to reflect the new extension names, making it easier for you to find the right extension for your project.

Reviewer: ```suggestion
- **Documentation:** We'll update our documentation to reflect the new extension names, making finding the right extension for your project easier.
```

> For current projects we recommend you to keep using the old names if you want to somehow still stay buildable against older version of Quarkus - otherwise upgrade in your own pace for new names.

Reviewer: ```suggestion
For current projects, we recommend continuing to use the old names if you want to remain buildable against an older version of Quarkus — otherwise, upgrade at your own pace for new names.
```

For new projects we will default and recommend to the new names.

> == What This Means for You as an Extension Writer

Reviewer 2: ```suggestion
== What This Means for You as an Extension Author
```

> If you've developed a Quarkus extension, and you want to stay compatible with Quarkus LTS releases, we recommend you to keep the old names as aliases for the new names. This will ensure that your extension remains compatible with older Quarkus versions. Relocations solves this.

Reviewer: ```suggestion
If you've developed a Quarkus extension and want to stay compatible with Quarkus LTS releases, we recommend keeping the old names as aliases for the new ones. This will ensure that your extension remains compatible with older Quarkus versions. Relocations solve this.
```

> For extensions just targeting 3.9+ you should use the new names. Possibly make a separate branch for 3.8 and older to keep the old names.

Reviewer: ```suggestion
For extensions targeting 3.9+, you should use the new names. Possibly make a separate branch for 3.8 and older to keep the old names.
```
